src/tests/java/picard/illumina/ExtractIlluminaBarcodesTest.java

```diff
@@ -105,18 +105,35 @@ public void testSingleEndWithBarcodeAtStart() throws Exception {
     }
 
     @Test
+    public void testSingleEndWithBarcodeAtStartAndMolecularIndicies() throws Exception {
+        final MetricsFile<ExtractIlluminaBarcodes.BarcodeMetric, Integer> metricsFile = runIt(1, "8B4M21T");
+        Assert.assertEquals(metricsFile.getMetrics().get(11).PERFECT_MATCHES, 1);
+    }
+
+    @Test
     public void testSingleEndWithBarcodeAtEnd() throws Exception {
         final MetricsFile<ExtractIlluminaBarcodes.BarcodeMetric, Integer> metricsFile = runIt(1, "25T8B");
         Assert.assertEquals(metricsFile.getMetrics().get(0).PERFECT_MATCHES, 5);
     }
 
     @Test
+     public void testSingleEndWithBarcodeAtEndAndMolecularIndicies() throws Exception {
+        final MetricsFile<ExtractIlluminaBarcodes.BarcodeMetric, Integer> metricsFile = runIt(1, "4M21T8B");
+        Assert.assertEquals(metricsFile.getMetrics().get(0).PERFECT_MATCHES, 5);
+    }
+
+    @Test
     public void testPairedEndWithBarcodeOnFirstEnd() throws Exception {
         final MetricsFile<ExtractIlluminaBarcodes.BarcodeMetric, Integer> metricsFile = runIt(1, "25T8B25T");
         Assert.assertEquals(metricsFile.getMetrics().get(0).PERFECT_MATCHES, 5);
     }
 
     @Test
+    public void testPairedEndWithBarcodeOnFirstEndAndMolecularIndicies() throws Exception {
+        final MetricsFile<ExtractIlluminaBarcodes.BarcodeMetric, Integer> metricsFile = runIt(1, "4M21T8B21T4M");
+        Assert.assertEquals(metricsFile.getMetrics().get(0).PERFECT_MATCHES, 5);
+    }
+    @Test
     public void testPairedEndWithBarcodeOnSecondEnd() throws Exception {
         final MetricsFile<ExtractIlluminaBarcodes.BarcodeMetric, Integer> metricsFile = runIt(1, "25T25T8B");
         Assert.assertEquals(metricsFile.getMetrics().get(12).PERFECT_MATCHES, 1);
```

Comment:
The molecular barcode isn't associated with either "end". in a sam record
it goes in all the records from that cluster, and in a fastq it is placed
in a file on its own....not sure this makes things more clear though...


On Sun, Nov 15, 2015 at 9:55 PM, Nils Homer <notifications@github.com>
wrote:

> In src/tests/java/picard/illumina/ExtractIlluminaBarcodesTest.java
> <https://github.com/broadinstitute/picard/pull/329#discussion_r44885142>:
>
> >      public void testPairedEndWithBarcodeOnFirstEnd() throws Exception {
> >          final MetricsFile<ExtractIlluminaBarcodes.BarcodeMetric, Integer> metricsFile = runIt(1, "25T8B25T");
> >          Assert.assertEquals(metricsFile.getMetrics().get(0).PERFECT_MATCHES, 5);
> >      }
> >
> >      @Test
> > +    public void testPairedEndWithBarcodeOnFirstEndAndMolecularIndicies() throws Exception {
> > +        final MetricsFile<ExtractIlluminaBarcodes.BarcodeMetric, Integer> metricsFile = runIt(1, "4M21T8B21T4M");
> > +        Assert.assertEquals(metricsFile.getMetrics().get(0).PERFECT_MATCHES, 5);
> > +    }
> > +    @Test
> >      public void testPairedEndWithBarcodeOnSecondEnd() throws Exception {
> >          final MetricsFile<ExtractIlluminaBarcodes.BarcodeMetric, Integer> metricsFile = runIt(1, "25T25T8B");
> >          Assert.assertEquals(metricsFile.getMetrics().get(12).PERFECT_MATCHES, 1);
>
> How would 25T8B4M17T be handled? Is 4M associated with the first end or
> second end?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/broadinstitute/picard/pull/329/files#r44885142>.
>